articles/115_topic_and_service_name_mapping.md

```diff
@@ -0,0 +1,313 @@
+Here is a table with some example expansions:
+
+| **Input Name** | Node: `my_node` NS: none | Node: `my_node` NS: `/foo`     |
+|----------------|--------------------------|--------------------------------|
+| `ping`         | *`/ping`*                | *`/foo/ping`*                  |
+| `/ping`        | *`/ping`*                | *`/ping`*                      |
+| `~`            | *`/my_node`*             | *`/foo/foo/my_node`*           |
+| `~/ping`       | *`/my_node/ping`*        | *`/foo/foo/my_node/ping`*      |
+| `/~`           | *`/my_node`*             | *`/foo/my_node`*               |
+| `/~/ping`      | *`/my_node/ping`*        | *`/foo/my_node/ping`*          |
+| `ping/~/pong`  | *`/ping/my_node/pong`*   | *`/foo/ping/foo/my_node/pong`* |
+| `/ping/~/pong` | *`/ping/my_node/pong`*   | *`/ping/foo/my_node/pong`*     |
+
+### Hidden Topic or Service Names
+
+Any topic or service name that contains any tokens (either namespaces or a topic or service name) which starts with an underscore (`_`) will be considered hidden and tools may not show them unless explicitly asked.
+
+## Mapping of ROS 2 Topic and Service Names to DDS Topics
+
+The ROS topic and service name constraints allow more characters than the DDS topic names because ROS allows for the forward slash (`/`) and the tilde (`~`).
+Since ROS 2 topic and service names are expanded to absolute names before being used, there is always a leading forward slash (`/`) and any tildes (`~`) will have been expanded.
+Additionally any URL related syntax is removed, e.g. the `rostopic://` prefix.
+Therefore the only forward slash has to be substituted when converting to DDS topic names.
+
+### ROS Specific Name Prefix
+
+In order to differentiate ROS topics easily, all DDS topic names created by ROS shall be prefixed with `rX_`, where `X` is a single character which indicates what subsystem of ROS to which the topic belongs.
+For example, a plain topic called `/foo` would translate to a DDS topic named `rt_foo`.
+
+For systems where Services are implemented with topics (like with OpenSplice), a different subsystem character can be used: `rq_` for the request topic and `rp_` for the response topic.
+On systems where the implementation is handled for us by DDS (like with Connext), we use `rS_` as the common prefix.
+
+Here is a non-exhaustive list of prefixes:
+
+| ROS Subsystem        | Prefix |
+|----------------------|--------|
+| ROS Topics           | rt_    |
+| ROS Service Request  | rq_    |
+| ROS Service Response | rs_    |
+| ROS Service          | rS_    |
+| ROS Parameter        | rp_    |
+| ROS Action           | ra_    |
+
+### Substitution of the Namespace Delimiter
+
+The namespace delimiter in ROS 2 topic and service names, a forward slash (`/`), will be replaced with double underscores (`__`).
+The leading forward slash (`/`) is not replaced with double underscores (`__`), and is replaced instead by the ROS specific name prefix.
```

Comment:
Imo keeping the parsing simpler and also replacing the leading `/` with two underscores outweighs the "benefit" of shortening the name by a single character.